src/scanspec/specs.py

```diff
@@ -164,14 +157,12 @@ class Repeat(Spec[Axis]):
     .. note:: There is no turnaround arrow at x=4
     """
 
-    num: ANum
-    gap: A[
-        bool,
-        schema(
-            description="If False and the slowest of the stack of Frames is snaked "
-            "then the end and start of consecutive iterations of Spec will have no gap"
-        ),
-    ] = True
+    num: int = Field(min=1, description="Number of frames to produce")
```

Comment:
Can we create Field instances up front and use them in two separate dataclasses?

Reply:
Probably, but there have been issues with doing so in the past (https://github.com/pydantic/pydantic/issues/1412), so I wanted to get this working without first of all.